pkg/kubelet/cm/container_manager_stub.go

```diff
@@ -43,12 +43,14 @@ import (
 type containerManagerStub struct {
 	shouldResetExtendedResourceCapacity bool
 	extendedPluginResources             v1.ResourceList
+	memoryManager                       memorymanager.Manager
 }
 
 var _ ContainerManager = &containerManagerStub{}
 
-func (cm *containerManagerStub) Start(_ context.Context, _ *v1.Node, _ ActivePodsFunc, _ GetNodeFunc, _ config.SourcesReady, _ status.PodStatusProvider, _ internalapi.RuntimeService, _ bool) error {
+func (cm *containerManagerStub) Start(ctx context.Context, _ *v1.Node, _ ActivePodsFunc, _ GetNodeFunc, _ config.SourcesReady, _ status.PodStatusProvider, _ internalapi.RuntimeService, _ bool) error {
 	klog.V(2).InfoS("Starting stub container manager")
+	cm.memoryManager = memorymanager.NewFakeManager(ctx)
```

Comment:
this is the exact same dilemma I'm facing with porting `cpumanager` to contextual logging.
This approach makes sense, but I can't help but ask myself (and others!) if the `New` operation should ideally take a context or if this is a byproduct of our porting, and if the latter is something we are willing to accept.

Reply:
Thanks for bringing this up. I found myself struggling a bit here as well. Ultimately, I took inspiration from your CPU manager PR in this area as I noticed no objections were raised in this area (and because I want to ensure consistency across the managers) 

Let's see if @bartosh has an opinion on this.

Reply:
I'm ok with this approach. However, there is an opinion that function should return an error if it takes context as a parameter. I'm not sure it makes sense in this particual case though.

Reply:
Yes, in this particular case, I don't see an immediate need for error handling since the function doesn’t perform operations that are expected to fail due to context constraints.